src/groundlight/internalapi.py

```diff
@@ -79,7 +150,40 @@ class GroundlightApiClient(ApiClient):
     """
 
     REQUEST_ID_HEADER = "X-Request-Id"
+    # This represents the byte stream of the input image file.
+    # We cache the byte stream so that if we attempt to access
+    # the file again we can do so even when the file has already been closed.
+    CACHED_BYTE_STREAM = io.BytesIO()
+
+    @staticmethod
+    def get_file_data_and_close_file(file_instance: io.IOBase) -> bytes:
+        """Overrides the method from OpenAPI client in order to allow access to the
+        file byte stream even after the file has been closed.
+        NOTE: Assumes that file_instance is either io.BytesIO or io.BufferedReader
+        """
+        if file_instance.closed:
```

Comment:
This logic isn't reliable - if the customer happens to pass in a file object which is already closed, then we'll do the wrong thing and re-send the previous image.

Reply:
Good catch!